src/components/molecules/Visualizer/WidgetAlignSystem/hooks.ts

```diff
@@ -0,0 +1,113 @@
+import { useCallback, useEffect, useState } from "react";
+import { Widget as WidgetType } from "../Widget";
+
+export type Location = {
+  zone?: string;
+  section?: string;
+  area?: string;
```

Comment:
Q: what values will come to here?

Reply:
A: This is the Location type that is used in a few different places. 
It is used for knowing where a widget is, and where it drops (in the onMove function).
It is used for the defaultLocation each widget has.
Etc etc.